Sentiment UI

HotLemonJuice opened the pull request:
To show the sentiment score on the map.

Details:
- Use random number for 0 - 4 to fake sentiment score for now
- Use sentiment score as the `count` for each polygon
- Disable the normalization toggle and set it to off when the sentiment toggle is on because it makes no sense to normalize sentiment scores
![screen shot 2017-04-26 at 15 20 06](https://cloud.githubusercontent.com/assets/12385178/25460649/9ca327b4-2a99-11e7-82bd-293de46f4198.jpg)

JavierJia commented on neo/public/javascripts/map/controllers.js:
```diff
@@ -371,19 +373,31 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common'])
+            return sentimentColors[2];
+          }
+        }
+        else{
```
```
} else {
```

JavierJia commented on neo/public/javascripts/map/controllers.js:
```diff
@@ -371,19 +373,31 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common'])
-          d = Math.ceil(Math.log10(d));
-          if(d <= 0) // treat smaller counts the same as 0
-            d = 0
+        if($scope.doSentiment){  // 0 <= d <= 4
```
could you make two functions `getSentimentColor` and `getCountColor`, then make this `getColor` more simpler?

JavierJia commented on neo/public/javascripts/map/controllers.js:
```diff
@@ -444,10 +458,17 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common'])
+                  geo['properties']['count'] = Math.random() * 4;
+                  geo["properties"]["countText"] = geo["properties"]["count"].toFixed(1);
+                }
+                else{
```
`} else {`

JavierJia commented on neo/public/javascripts/map/controllers.js:
```diff
@@ -471,44 +496,56 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common'])
+          div.innerHTML +=
+            '<i style="background:' + getColor(3) + '"></i>Positive<br>';
         }
+        else{
```
`} else {`

chenlica commented:
Since this PR is adding a new "face" to the UI, it's time to think about how to allow the users to choose the different faces.  One possible design is to provide a few options such as "Density, Normalized Density, Sentiment".  Based on the choice, we show the corresponding face.  Thoughts?

HotLemonJuice commented:
@JavierJia I have addressed your comments and refactored the code a lot, especially in the legend/normalizeToggle/sentimentToggle part. Can you review the new changes?

HotLemonJuice commented:
@chenlica Currently, we use toggles to let users switch between faces. However, I found the sidebar from the MapD example could be a better option to list different options for the UI for the following reasons:
(1) It takes less space
(2) We can potentially list more options than stacking them 
![screen shot 2017-04-27 at 10 52 42](https://cloud.githubusercontent.com/assets/12385178/25497004/f453699e-2b37-11e7-964a-b3adae481611.jpg)
![screen shot 2017-04-27 at 10 52 50](https://cloud.githubusercontent.com/assets/12385178/25497009/f74dd4ae-2b37-11e7-85e4-e8d75804993a.jpg)

chenlica commented:
@HotLemonJuice We can use something similar to their UI.  In general, we expect to add various analysis functionalities to the UI, so it's important to allow the user to select different "faces."

JavierJia commented on neo/public/javascripts/map/controllers.js:
```diff
@@ -71,7 +72,8 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common'])
         },
-        colors: [ '#f7f7f7', '#92c5de', '#4393c3', '#2166ac', '#f4a582', '#d6604d', '#b2182b']
+        colors: [ '#ffffff', '#92c5de', '#4393c3', '#2166ac', '#f4a582', '#d6604d', '#b2182b'],
+        sentimentColors: ['#ff0000', '#C0C0C0', '#00ff00']
```
👍

JavierJia commented on neo/public/javascripts/map/controllers.js:
```diff
@@ -444,10 +464,16 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common'])
+                if($scope.doSentiment){
+                  // TODO: change fake random number to real sentiment (0-4)
+                  geo['properties']['count'] = Math.random() * 4;
+                  geo["properties"]["countText"] = geo["properties"]["count"].toFixed(1);
```
should we regulate the string format? do you know which is norm? `"` or `'`?

HotLemonJuice replied:
As often indicated by the codacy, double quote is the norm to follow. There are many places in the project involving double quotes and single quotes. I think we will need a dedicated PR to fix it.

chenlica commented:
@JavierJia : OK, let's merge this one, then do the improvements step by step.
